holoviews/element/raster.py

```diff
@@ -723,19 +746,27 @@ def __getitem__(self, coords):
         Slice the underlying numpy array in sheet coordinates.
         """
         if coords in self.dimensions(): return self.dimension_values(coords)
+        coords = util.process_ellipses(self, coords)
         if not isinstance(coords, slice) and len(coords) > self.ndims:
-            value = coords[self.ndims:]
-            if len(value) > 1:
-                raise KeyError("Only one value dimension may be indexed at a time")
-
-            sliced = super(RGB, self).__getitem__(coords[:self.ndims])
-            vidx = self.get_dimension_index(value[0])
-            val_index = vidx - self.ndims
-            data = sliced.data[:,:, val_index]
-            return Image(data, **dict(self.get_param_values(onlychanged=True),
-                                       vdims=[self.vdims[val_index]]))
-        else:
-            return super(RGB, self).__getitem__(coords)
+            values = coords[self.ndims:]
+            channels = [el for el in values if isinstance(el, (str, Dimension))]
+            if len(channels) == 1:
+                sliced = super(RGB, self).__getitem__(coords[:self.ndims])
+                if channels[0] not in self.vdims:
+                    raise KeyError("%r is not an available value dimension"
+                                    % channels[0])
+                vidx = self.get_dimension_index(channels[0])
+                val_index = vidx - self.ndims
+                data = sliced.data[:,:, val_index]
+                return Image(data, **dict(self.get_param_values(onlychanged=True),
```

Comment:
Not your fault as this was already wrong but this isn't quite right, could use core.util.get_param_values instead of self.get_param_values here.